mqtt/mqtt-broker/tests/broker_model.rs

```diff
@@ -0,0 +1,240 @@
+use std::collections::{HashMap, HashSet};
+
+use proptest::{prop_oneof, proptest, strategy::Strategy};
+use tokio::sync::mpsc::{self, UnboundedReceiver};
+use uuid::Uuid;
+
+use mqtt3::proto;
+use mqtt_broker::{
+    proptest::{arb_client_id_weighted, arb_connect, arb_subscribe, arb_unsubscribe},
+    AuthId, BrokerBuilder, ClientEvent, ClientId, ConnReq, ConnectionHandle, Message,
+};
+
+proptest! {
+    #[test]
+    fn broker_manages_sessions(
+        events in proptest::collection::vec(arb_broker_event(), 1..50)
+    ) {
+    tokio::runtime::Builder::new()
+        .basic_scheduler()
+        .enable_all()
+        .build()
+        .unwrap()
+        .block_on(test_broker_manages_sessions(events));
+    }
+}
+
+async fn test_broker_manages_sessions(events: impl IntoIterator<Item = BrokerEvent>) {
```

Comment:
I think this module would benefit from code comments. It would be easier to follow for those who new or unfamiliar with model-base testing. Even for those who is familiar with it, it would be easier to understand how your model works and how events flow in the model.